connector/src/main/java/org/geysermc/connector/network/UpstreamPacketHandler.java

```diff
@@ -138,4 +183,16 @@ public boolean handle(MovePlayerPacket packet) {
     boolean defaultHandler(BedrockPacket packet) {
         return translateAndDefault(packet);
     }
+
+    @Override
+    public boolean handle(DisconnectPacket packet) {
+        final int clientId = packet.getClientId();
+        GeyserSession session = sessions.get(clientId);
+
+        session.disconnect(packet.getKickMessage());
+
+        this.sessions.remove(clientId);
+
```

Comment:
Need to test if we should only handle SubClients here and just pass the packet along for the MainClient. Because that flow is already handled elsewhere.

Reply:
Changed it for now, just in case.